src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -2217,7 +2217,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     ====================================================================================================
   -->
   <Target Name="GenerateBindingRedirects"
-    Inputs="$(MSBuildAllProjects);@(AppConfigFile);$(ResolveAssemblyReferencesStateFile);$(IntermediateOutputPath);@(SuggestedBindingRedirects)"
+    Inputs="$(MSBuildNewestProject);$(MSBuildAllProjects);@(AppConfigFile);$(ResolveAssemblyReferencesStateFile);$(IntermediateOutputPath);@(SuggestedBindingRedirects)"
```

Comment:
For targets we control, do we need to pass `$(MSBuildNewestProject);$(MSBuildAllProjects)`? That's what I'd expect for someone who wanted to support MSBuild 15 and MSBuild 16, but in our targets where we'll only run on 16+ I think we should switch instead of passing an additional file. That way we get the benefit of not having to `stat` all the project files.

Reply:
I agree, we should never use `MSBuildAllProjects` as an Input in our targets that ship with MSBuild again. Setting it is a different story, but shouldn't consume it.

Reply:
After discussing this, it might better to not set it and leave it in the `Inputs`.  That way if other people are adding to it we won't break them.  They would need to remove the instance of setting the `MSBuildAllProjects` property if its just an import but if its not an import the behavior would stay the same.